sdks/go/pkg/beam/coder.go

```diff
@@ -85,18 +85,31 @@ func inferCoder(t FullType) (*coder.Coder, error) {
 	switch t.Class() {
 	case typex.Concrete, typex.Container:
 		switch t.Type() {
-		// The type conversions here are very conservative. We handle bytes/strings
-		// equivalently because they are essentially equivalent in the language.
-		// Notably, we do not (currently) support equivalences in numeric data types
-		// due to risks around inadvertent widening or narrowing of data.
+		case reflectx.Int, reflectx.Int8, reflectx.Int16, reflectx.Int32, reflectx.Int64:
+			c, err := coder.NewVarIntZ(t.Type())
+			if err != nil {
+				return nil, err
+			}
+			return &coder.Coder{Kind: coder.Custom, T: t, Custom: c}, nil
+		case reflectx.Uint, reflectx.Uint8, reflectx.Uint16, reflectx.Uint32, reflectx.Uint64:
+			c, err := coder.NewVarUintZ(t.Type())
+			if err != nil {
+				return nil, err
+			}
+			return &coder.Coder{Kind: coder.Custom, T: t, Custom: c}, nil
 		case reflectx.String, reflectx.ByteSlice:
+			// The type conversions here are very conservative. We handle bytes/strings
+			// equivalently because they are essentially equivalent in the language.
+			// Notably, we do not (currently) support equivalences in numeric data types
```

Comment:
This part of the comment would go better with the int conversion code.

Do we want to add support for encoding floats? Some applications associate a weight or score with a value that is float-y, so supporting that easily would be helpful.

Reply:
Rephrased the whole comment. It seemed a bit off, because there is no a priori motivation to pick a coder that needs runtime conversions.

I'm not opposed to be adding floats in a separate PR. Is there a standard binary encoding we can use?